### ISSUE TITLE
A failed dolt sql -q "SELECT DOLT_BRANCH" sometimes return an unexpected sql row in the output

### ISSUE THREAD

**Issue by druvv** [dolthub/dolt#3357]
I first found this failure in a bats test. When running `sql-branch.bats`, test "DOLT_BRANCH -c throws error on error cases"  sometimes failed on an output check.

After running the test in a loop (for hundreds of times), we occasionally see this output from
`dolt sql -q "SELECT DOLT_BRANCH('-c','','copy')` and any variation of the parameters.
```
 +-----------------------------+ | DOLT_BRANCH('-c','','copy') | +-----------------------------+ +-----------------------------+
 fatal: A branch named 'original' not found
```

The expected output is just:
```
fatal: A branch named 'original' not found
```

There seems to be a race issue with how the error is being handled and how it is being printed in the results.

**Comment by druvv** [comment]
> Theory is it comes from some racey interaction between the pipeline printer and the error that the function generates.

@reltuk's theory

**Comment by druvv** [comment]
fixed in #3443
